plugins/azure-devops-backend/src/api/PullRequestsDashboardProvider.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DashboardPullRequest,
+  PullRequestOptions,
+  Team,
+  TeamMember,
+} from '@backstage/plugin-azure-devops-common';
+
+import { AzureDevOpsApi } from './AzureDevOpsApi';
+import { Logger } from 'winston';
+
+export class PullRequestsDashboardProvider {
+  private teams = new Map<string, Team>();
+
+  private teamMembers = new Map<string, TeamMember>();
+
+  private constructor(
+    private readonly logger: Logger,
+    private readonly azureDevOpsApi: AzureDevOpsApi,
+  ) {}
+
+  public static async create(
+    logger: Logger,
+    azureDevOpsApi: AzureDevOpsApi,
+  ): Promise<PullRequestsDashboardProvider> {
+    const provider = new PullRequestsDashboardProvider(logger, azureDevOpsApi);
+    await provider.readTeams();
+    return provider;
+  }
+
+  public async readTeams(): Promise<void> {
+    this.logger.info('Reading teams.');
+
+    let teams = await this.azureDevOpsApi.getAllTeams();
+
+    // This is used to filter out the default Azure Devops project teams.
+    teams = teams.filter(team =>
+      team.name && team.projectName
+        ? team.name !== `${team.projectName} Team`
+        : true,
+    );
+
+    this.teams = new Map<string, Team>();
+    this.teamMembers = new Map<string, TeamMember>();
+
+    await Promise.all(
+      teams.map(async team => {
```

Comment:
this code should probably be wrapped in a `p-limit` of a reasonable size (like 3, or so) to reduce the unlimited parallelism introduced here - search through the repo for some prior usages if you want

Reply:
Thanks. I've implemented a `p-limit`.